statistics/selectivity_test.go

```diff
@@ -593,3 +593,25 @@ func (s *testStatsSuite) TestUniqCompEqualEst(c *C) {
 		testKit.MustQuery(input[i]).Check(testkit.Rows(output[i]...))
 	}
 }
+
+func (s *testStatsSuite) TestCascadePlannerHashedPartTable(c *C) {
```

Comment:
Could we move this test case to `planner/cascades/integration_test.go`?

Reply:
Hi,
I mean `planner/cascades/integration_test.go`, not `planner/core/integration_test.go`. Because currently, we uses the package `planner/cascades` to develop the cascades planner. Since this bug-fix is related to the cascades planner, we think we'd better move it there. 
BTW, I think we don't need to check the query plan when `tidb_enable_cascades_planner=0`. But we need to check whether this query(`select * from pt`) can get all of the rows of the partitioned table.

Reply:
Oh I misread it, Thanks. Will move it to right place.

I will add the select query to get all the rows(will ensure that number of  rows>partitions) after enabling cascade planner. I added disable case as well to test existing also works.
Let me know if you have other ideas.

Reply:
> Hi,
> I mean `planner/cascades/integration_test.go`, not `planner/core/integration_test.go`. Because currently, we uses the package `planner/cascades` to develop the cascades planner. Since this bug-fix is related to the cascades planner, we think we'd better move it there.

I have made necessary changes.

> BTW, I think we don't need to check the query plan when `tidb_enable_cascades_planner=0`. But we need to check whether this query(`select * from pt`) can get all of the rows of the partitioned table.

Right now, it returns empty result even though there are rows present in the table. 
During my investigations, I found out https://github.com/pingcap/tidb/blob/2ee9964b51908960a47550a9d9b41d189c52d484/planner/core/logical_plan_builder.go#L2510-L2511
`statsHandle.statsCache.tables[pid]` returns `nil`. However, all 4 partition's information were present. 
E.g.
Let's assume that `physicalID` for table `pt1` is `96` & it has 4 partitions`97`,`98`,`99` & `100`  as their respective `physicalID`s. The key in map `statsHandle.statsCache.tables` only contained `97`,`98`,`99` & `100` but not `96`.

To fix this issue, it may require many changes. It would be better to handle the empty result bug separately from this bug.

Reply:
Sure, the support of partitioned table in cascades planner requires a lot of changes, like what we have done in `planner/core/rule_partition_processor.go`. IMHO, currently we could enhance the rule `EnumeratePaths` to handle partition table.

Reply:
Can you please share the design document for new planner? I could get only https://github.com/pingcap/tidb/blob/master/docs/design/2018-08-29-new-planner.md